rcl_lifecycle/include/rcl_lifecycle/rcl_lifecycle.h

```diff
@@ -430,6 +430,25 @@ RCL_LIFECYCLE_PUBLIC
 void
 rcl_print_state_machine(const rcl_lifecycle_state_machine_t * state_machine);
 
+/// Print the transition map
+/**
+ * This function will print in the standard output the transition map
+ * of the rcl_lifecycle_state_machine_t struct.
+ *
+ * <hr>
+ * Attribute          | Adherence
+ * ------------------ | -------------
+ * Allocates Memory   | No
```

Comment:
I think it does allocate memory as the logging functions that do formatting will need to allocate in some cases I think.